### PR TITLE
Tests: 1175 ci refactor e2e

### PR DESCRIPTION
**Pull request by bermuell**
## Description

Closes: #1175

- Created github workflow e2e tests
- changed e2e tests for easier test selection (see -tc argument)
- dedicated jobs for each e2e test case on manual and nightly workflow
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: .github/workflows/manual-e2e.yml [hunk 1]
```diff
@@ -5,22 +5,99 @@ on:
...
+          go-version: "1.20"
+      - uses: actions/checkout@v3
+      - name: Checkout LFS objects
+        run: git lfs checkout
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20" # The Go version to download (if necessary) and use.
+      - name: E2E changeover test
+        run: go run ./tests/e2e/... --tc changeover
```
Comment (MSalopek) [inline .github/workflows/manual-e2e.yml:39]: Looks good, can you also check `Makefile` commands?

If you have time, consider refactoring the Makefile commands listed below. You're free to change or remove the commands as you see fit:
```Makefile

# run only happy path E2E tests
test-e2e-short:
	go run ./tests/e2e/... --happy-path-only

# run only happy path E2E tests with cometmock
# this set of traces does not test equivocation but it does check downtime
test-e2e-short-cometmock:
	go run ./tests/e2e/... --cometmock-happy-path --use-cometmock --use-gorelayer

# run full E2E tests in sequence (including multiconsumer)
test-e2e-multi-consumer:
	go run ./tests/e2e/... --include-multi-consumer

# run full E2E tests in parallel (including multiconsumer)
test-e2e-parallel:
	go run ./tests/e2e/... --include-multi-consumer --parallel

# run full E2E tests in sequence (including multiconsumer) using latest tagged gaia
test-gaia-e2e:
	go run ./tests/e2e/... --include-multi-consumer --use-gaia

# run only happy path E2E tests using latest tagged gaia
test-gaia-e2e-short:
	go run ./tests/e2e/... --happy-path-only --use-gaia

# run full E2E tests in parallel (including multiconsumer) using latest tagged gaia
test-gaia-e2e-parallel:
	go run ./tests/e2e/... --include-multi-consumer --parallel --use-gaia

# run full E2E tests in sequence (including multiconsumer) using specific tagged version of gaia
# usage: GAIA_TAG=v9.0.0 make test-gaia-e2e-tagged
test-gaia-e2e-tagged:
	go run ./tests/e2e/... --include-multi-consumer --use-gaia --gaia-tag $(GAIA_TAG)

# run only happy path E2E tests using latest tagged gaia
# usage: GAIA_TAG=v9.0.0 make test-gaia-e2e-short-tagged
test-gaia-e2e-short-tagged:
	go run ./tests/e2e/... --happy-path-only --use-gaia --gaia-tag $(GAIA_TAG)

# run full E2E tests in parallel (including multiconsumer) using specific tagged version of gaia
# usage: GAIA_TAG=v9.0.0 make test-gaia-e2e-parallel-tagged
test-gaia-e2e-parallel-tagged:
	go run ./tests/e2e/... --include-multi-consumer --parallel --use-gaia --gaia-tag $(GAIA_TAG)

```

Comment (bermuell) [reply .github/workflows/manual-e2e.yml:39]: Makefile should be adapted already to the  changes

**Review by MSalopek — APPROVED** [review]
Approval!

Thanks for spending the time to refactor this!

**Comment by smarshall-spitzbart** [comment]
My understanding is that the changed files `manual-e2e.yml` and `nightly-e2e.yml` are not used for CI. Rather `automated-tests.yml` is used for CI. The issue seems to mention improving CI reporting, which would be unchanged by this PR due to `automated-tests.yml` being unchanged. Am I misunderstanding something here?

**Comment by bermuell** [comment]
> My understanding is that the changed files `manual-e2e.yml` and `nightly-e2e.yml` are not used for CI. Rather `automated-tests.yml` is used for CI. The issue seems to mention improving CI reporting, which would be unchanged by this PR due to `automated-tests.yml` being unchanged. Am I misunderstanding something here?

Reason is that the changes forseen for the automated tests (splitting E2E from UT, IT) were addressed by PR #1181 and introduction of cometmock based tests changed the need for spliting happy path test in respect of downtime (normal vs throttling).

**Review by p-offtermatt — APPROVED** [review]
LGTM. I pushed one change to the docstring of the short happy path.

I also think we might want to rename the short happy path to cometmock-happy-path or something like this, since it's getting longer and really the main distinction is that it avoids things incompatible with cometmock specifically. But this is not really that important imo, so I'm happy with this as-is, too

**Pull request merged**